### PR TITLE
BUG: fillna returns frame when inplace=True if value is a dict (#16156)

### LINKED ISSUE #16156: BUG: fillna returns frame when inplace=True if fill values are specified as a dict
`fillna` will return the frame in question if passed fill values via a dict, even if `inplace=True`.

```
In [6]: data = pd.DataFrame({'a': [1, None], 'b': [2, None]})

In [7]: data.fillna({'a': 0}, inplace=True)
Out[7]: 
     a    b
0  1.0  2.0
1  0.0  NaN

In [8]: data
Out[8]: 
     a    b
0  1.0  2.0
1  0.0  NaN

```

### PR DESCRIPTION
**Pull request by ante012**
- [x] closes #16156
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.21.0.txt [hunk 1]
```diff
@@ -391,3 +391,4 @@ Other
 - Bug in :func:`eval` where the ``inplace`` parameter was being incorrectly handled (:issue:`16732`)
 - Bug in ``.isin()`` in which checking membership in empty ``Series`` objects raised an error (:issue:`16991`)
 - Bug in :func:`unique` where checking a tuple of strings raised a ``TypeError`` (:issue:`17108`)
+- Bug in :func: `fillna` returns frame when ``inplace=True`` and ``value`` is dict (:issue:`16156`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.21.0.txt]: `` :func:`Series.fillna` ``

move to conversions.

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. doc comment.

File: doc/source/whatsnew/v0.21.0.txt (hunk 1 above)
Comment (ante328) [inline doc/source/whatsnew/v0.21.0.txt]: done

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**